pkg/install/check/check.go

```diff
@@ -0,0 +1,525 @@
+	"github.com/acorn-io/baaah/pkg/restconfig"
+	authorizationv1 "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+	klogv2 "k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CheckResult describes the results of a check, making it human-readable
+type CheckResult struct {
+	Message string `json:"message"`
+	Passed  bool   `json:"passed"`
+	Name    string `json:"name"`
+}
+
+// PreflightChecks is a list of all checks that are run before the installation.
+// They are crictial and will make the installation fail.
+func PreflightChecks(ctx context.Context) []CheckResult {
+	return RunChecks(ctx, CheckRBAC, CheckNodesReady)
+}
+
+// InFlightChecks is a list of all checks that are run after the installation.
+// They are not critical and should not affect the installation process.
+func InFlightChecks(ctx context.Context) []CheckResult {
+	checks := []func(ctx context.Context) CheckResult{
+		CheckDefaultStorageClass,
+		CheckIngressCapability,
+		CheckExec,
+	}
+
+	// Some debugging test
+	if os.Getenv("ACORN_INSTALL_FAIL_CHECKS") == "true" {
+		checks = append(checks, func(ctx context.Context) CheckResult {
+			return CheckResult{Name: "FailTest", Passed: false, Message: "This is a test failure"}
+		})
+	}
+
+	return RunChecks(ctx, checks...)
+}
+
+// IsFailed is a simple helper function marking a list of check results
+// as failed if one or more results show failed status.
+func IsFailed(results []CheckResult) bool {
+	for _, r := range results {
+		if !r.Passed {
+			return true
+		}
+	}
+	return false
+}
+
+// RunChecks runs a list of checks and returns their results as a list.
+func RunChecks(ctx context.Context, checks ...func(ctx context.Context) CheckResult) []CheckResult {
+	var results []CheckResult
+	for _, check := range checks {
+		results = append(results, check(ctx))
+	}
+	return results
+}
+
+// silenceKlog is a helper function to silence klog output which could disturb
+// the clean installation logs.
+func silenceKlog() {
+	klog.SetOutput(io.Discard)
+	klogv2.SetOutput(io.Discard)
+	utilruntime.ErrorHandlers = nil
+}
+
+func CheckExec(ctx context.Context) CheckResult {
+	result := CheckResult{
+		Name: "Exec",
+	}
+
+	silenceKlog()
+	cli, err := k8sclient.Default()
+	if err != nil {
+		result.Passed = false
+		result.Message = fmt.Sprintf("Error creating client: %v", err)
+		return result
+	}
+
+	// Create new pod
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "acorn-install-check-exec",
+			Namespace: system.Namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:            "exec",
+					Image:           "busybox",
```

Comment:
could present an issue for airgap installs.

Reply:
would using the acorn image be better? ...of course then you need to know the version of the image thats available

Reply:
This is how it is retrieved/got during install:
https://github.com/acorn-io/acorn/blob/main/pkg/cli/install.go#L35-L38

You don't want to just use the defaultImage, you want to use the user-supplied one. So somehow you have to hook into that logic

Reply:
> @ibuildthecloud said that you should be able to find the image we are going to install. Should be an api for it.

I guess this won't work for the `acorn check` command, right?
Should I introduce a `--image` flag there? :thinking:

Reply:
I found it was the easiest to just pass in that information via an options struct :man_shrugging: 
If you have any better idea, let me know :)